Content.Shared/Chemistry/EntitySystems/SolutionContainerSystem.Capabilities.cs

```diff
@@ -0,0 +1,139 @@
+        {
+            if (owner.TryGetComponent(out InjectableSolutionComponent? injectable) &&
+                owner.TryGetComponent(out SolutionContainerManagerComponent? manager) &&
+                manager.Solutions.TryGetValue(injectable.Solution, out solution))
+            {
+                return true;
+            }
+
+            solution = null;
+            return false;
+        }
+
+        public bool TryGetRefillableSolution(IEntity owner,
+            [NotNullWhen(true)] out Solution? solution)
+        {
+            if (owner.TryGetComponent(out RefillableSolutionComponent? refillable) &&
+                owner.TryGetComponent(out SolutionContainerManagerComponent? manager) &&
+                manager.Solutions.TryGetValue(refillable.Solution, out solution))
+            {
+                return true;
+            }
+
+            solution = null;
+            return false;
+        }
+
+        public bool TryGetDrainableSolution(IEntity owner,
+            [NotNullWhen(true)] out Solution? solution)
+        {
+            if (owner.TryGetComponent(out DrainableSolutionComponent? drainable) &&
+                owner.TryGetComponent(out SolutionContainerManagerComponent? manager) &&
+                manager.Solutions.TryGetValue(drainable.Solution, out solution))
+            {
+                solution.OwnerUid = owner.Uid;
+                return true;
+            }
+
+            solution = null;
+            return false;
+        }
+
+        public bool TryGetDrawableSolution(IEntity owner,
+            [NotNullWhen(true)] out Solution? solution)
+        {
+            if (owner.TryGetComponent(out DrawableSolutionComponent? drawable) &&
+                owner.TryGetComponent(out SolutionContainerManagerComponent? manager) &&
+                manager.Solutions.TryGetValue(drawable.Solution, out solution))
+            {
+                solution.OwnerUid = owner.Uid;
+                return true;
+            }
+
+            solution = null;
+            return false;
+        }
+
+        public ReagentUnit DrainAvailable(IEntity? owner)
+        {
+            if (owner == null || !TryGetDrainableSolution(owner, out var solution))
+                return ReagentUnit.Zero;
+
+            return solution.CurrentVolume;
+        }
+
+        public bool HasFitsInDispenser(IEntity owner)
+        {
+            return !owner.Deleted && owner.HasComponent<FitsInDispenserComponent>();
+        }
+
+        public bool TryGetFitsInDispenser(EntityUid owner,
+            [NotNullWhen(true)] out Solution? solution)
+        {
+            if (_entityManager.TryGetEntity(owner, out var ownerEntity) &&
+                ownerEntity.TryGetComponent(out FitsInDispenserComponent? dispenserFits) &&
+                ownerEntity.TryGetComponent(out SolutionContainerManagerComponent? manager) &&
+                manager.Solutions.TryGetValue(dispenserFits.Solution, out solution))
```

Comment:
Generally better to use ComponentManager, especially if you already have the EntityUid, as it's faster than doing TryGetComponent